plugins/inputs/procstat/connections_linux.go

```diff
@@ -0,0 +1,400 @@
+
+			for _, proc := range inodeInfo {
+				if linux.TCPState(diag.State) == linux.TCP_LISTEN {
+					if !nsHandle.Equal(defaultNetNs) {
+						// Ignore listener outside of the default net NS
+						nsHandle.Close()
+						continue
+					}
+					listeners[uint32(diag.SrcPort())] = nil
+				}
+
+				connectionsByPid[proc.pid] = append(connectionsByPid[proc.pid], connInfo{
+					netNs:   netNs.id,
+					state:   linux.TCPState(diag.State),
+					srcIP:   diag.SrcIP(),
+					srcPort: uint32(diag.SrcPort()),
+					dstIP:   diag.DstIP(),
+					dstPort: uint32(diag.DstPort()),
+				})
+			}
+		}
+		nsHandle.Close()
+	}
+
+	return connectionsByPid, listeners, nil
+}
+
+// gatherData parse /proc to get procs info grouped by network namespace
+func (n *netNSProcInfo) gatherData() error {
+	n.data = map[string]netNSProcs{}
+
+	fd, err := os.Open("/proc")
+	if err != nil {
+		return fmt.Errorf("opening /proc: %w", err)
+	}
+	defer fd.Close()
+
+	dirContents, err := fd.Readdirnames(0)
+	if err != nil {
+		return fmt.Errorf("reading /proc: %w", err)
+	}
+
+	for _, pidStr := range dirContents {
+		n.readPidInfo(pidStr)
+	}
+
+	return nil
+}
+
+// readPidInfo for each PID, get network namespace and associated inodes
+func (n *netNSProcInfo) readPidInfo(pidStr string) {
+	pid, err := strconv.ParseUint(pidStr, 10, 32)
+	if err != nil {
+		// exclude files with a not numeric name. We only want to access pid directories
+		return
+	}
+
+	// Get process net namespace
+	netNs, err := os.Readlink("/proc/" + pidStr + "/ns/net")
+	if err != nil {
+		// ignore errors:
+		//   - missing directory, pid has already finished
+		//   - permission denied
+		return
+	}
+
+	pidDir, err := os.Open("/proc/" + pidStr + "/fd/")
+	if err != nil {
+		// ignore errors:
+		//   - missing directory, pid has already finished
+		//   - permission denied
+		return
+	}
+	defer pidDir.Close()
+
+	fds, err := pidDir.Readdirnames(0)
+	if err != nil {
+		// ignore errors:
+		//   - missing directory, pid has already finished
+		return
+	}
+
+	for _, fd := range fds {
+		link, err := os.Readlink("/proc/" + pidStr + "/fd/" + fd)
+		if err != nil {
+			// ignore errors:
+			//   - missing file, fd disappeared
+			continue
+		}
+
+		var procInode uint32
+
+		_, err = fmt.Sscanf(link, "socket:[%d]", &procInode)
+		if err != nil {
+			// this inode is not a socket
+			continue
+		}
+
+		n.AddProc(netNs, uint32(pid), inode(procInode))
+	}
+}
+
+// AddProc store pid with its associated network namespace an inode
+func (n *netNSProcInfo) AddProc(netNs string, pid uint32, procInode inode) {
+	netNsData, ok := n.data[netNs]
+	if !ok {
+		netNsData = netNSProcs{
+			id:          netNs,
+			inodeToProc: map[inode][]inodeInfo{},
+		}
+	}
+
+	procInfo, ok := netNsData.inodeToProc[procInode]
+	if ok {
+		procInfo = append(procInfo, inodeInfo{pid: pid})
+		netNsData.inodeToProc[procInode] = procInfo
+	} else {
+		netNsData.inodeToProc[procInode] = []inodeInfo{{pid: pid}}
+	}
```

Comment:
I think you can just do
```suggestion
	netNsData.inodeToProc[procInode] = append(netNsData.inodeToProc[procInode], inodeInfo{pid: pid})
```